doc/mobius/migrating-to-mobius.md

```diff
@@ -1,7 +1,99 @@
+5. If there are tests that have conditional assertion blocks, break them out into different tests (one for each conditional branch) that assert one idea each.
+6. Verify that as much coverage is present before proceeding to the next stage.
+   - Comment out **ONE** flow setup method in the `.apply()` method of the controller and run the tests and verify that some of the tests have broken.
+   - If no tests have broken, add tests to capture the existing flow.
+   - If some tests have broken, skim through them and verify that the current behaviour is captured completely. If it's not, add tests to capture it.
+   - Repeat the above steps for all the flow setup methods in the controller.
+7. If the controller accepts runtime dependencies via a custom `ScreenCreated` type, change the controller to accept the dependencies via `@AssistedInject` and remove the custom screen created event.
+
+### Strangulate
+This stage is where we break up the controller into the Mobius components. There are three sub-phases here:
+
+#### Prepare the controller
+1. Extract all the functions on the screen the controller invokes into a separate interface. Change the controller and the tests to only be aware of the interface and not the concrete screen type.
+2. Add all the Mobius classes into the package with no-op implementations.
+   - When creating the `Event` type, make it implement the `UiEvent` marker interface.
+   - When the creating the `Model` type, make it `Parcelable`.
+3. Set up the Mobius loop in the controller tests
+   -  Construct a `MobiusTestFixture` and set it up in the `@Before` test fixture. Dispose it in the `@After` fixture.
+   -  Add a method `startMobiusLoop` and invoke `MobiusTestFixture#start` in there.
+   -  Wherever the `setupController` method is being used, also add the `startMobiusLoop` method.
+4. Integrate the Mobius loop in the screen
+   -  Extract the event stream in the screen to a `lazy` delegated property.
+   -  Move the composition of the `ReportAnalyticsEvents` from the controller to this stream in the screen.
+   -  Share this stream using `Observable#share()`
+   -  Add the `MobiusDelegate` to the screen and hook into the lifecyle methods.
+5. If the UI interface includes one-off methods (navigating to another screen, showing a pop-up, etc), create a blank `UiActions` interface and implement this on the UI interface. This class should also be added as a dependency on the effect handler.
+
+Run all tests in the screen (from the package if the test class has been split into multiple) now to ensure that nothing has been broken. If it has, investigate and fix before proceeding.
+
+#### Migrate the logic
+> It is critical here to *NOT* attempt to refactor the effect handler or the Mobius update loop during the process of migration. If you see duplication when migrating the methods, let it remain there. We will clean up and refactor everything in the end after all the flows are moved.
```

Comment:
I love that you have added pointers like this to the doc. We can add more of these maybe when we pair on refactoring.